packages/@azure/eventhubs/client/lib/connectionContext.ts

```diff
@@ -63,6 +63,34 @@ export namespace ConnectionContext {
     return finalUserAgent;
   }
 
+  /**
+   * Removes all the sender links , receiver links, cbs session and management session to the Event Hub.
+   * @returns {void} void
+   */
+   function remove(connectionContext: ConnectionContext): void {
+    try {
+        // Remove all the senders link.
+       for (const senderName of Object.keys(connectionContext.senders)) {
```

Comment:
This in .NET would not actually work since you are modifying the collection while iterating over it -- @ShivangiReja, @ramya-rao-a can you explain to me how is this magic happening? :)

Reply:
The collection is not really being modified here. The `remove` on the sender or receiver object here then calls the `remove` on the underlying rhea links. It would be more appropriate to call the function `removeLink` to avoid the impression that the collection entries are being removed

Reply:
I'll update the method name to `removeLink`.